packages/plugin-ext/src/plugin/quick-open.ts

```diff
@@ -123,3 +130,123 @@ export class QuickOpenExtImpl implements QuickOpenExt {
+    private readonly onDidHideEmitter: Emitter<void>;
+    private readonly onDidAcceptEmitter: Emitter<void>;
+    private readonly onDidChangeActiveEmitter: Emitter<T[]>;
+    private readonly onDidChangeSelectionEmitter: Emitter<T[]>;
+    private readonly onDidChangeValueEmitter: Emitter<string>;
+    private readonly onDidTriggerButtonEmitter: Emitter<QuickInputButton>;
+
+    constructor(readonly quickOpen: QuickOpenExtImpl) {
+        this.buttons = [];
+        this.step = 0;
+        this.title = '';
+        this.totalSteps = 0;
+        this.value = '';
+        this.disposableCollection = new DisposableCollection();
+        this.disposableCollection.push(this.onDidHideEmitter = new Emitter());
+        this.disposableCollection.push(this.onDidAcceptEmitter = new Emitter());
+        this.disposableCollection.push(this.onDidChangeActiveEmitter = new Emitter());
+        this.disposableCollection.push(this.onDidChangeSelectionEmitter = new Emitter());
+        this.disposableCollection.push(this.onDidChangeValueEmitter = new Emitter());
+        this.disposableCollection.push(this.onDidTriggerButtonEmitter = new Emitter());
+    }
+
+    get items(): T[] {
+        return this._items;
+    }
+
+    set items(activeItems: T[]) {
+        this._items = activeItems;
+    }
+
+    get activeItems(): T[] {
+        return this._activeItems;
+    }
+
+    set activeItems(activeItems: T[]) {
+        this._activeItems = activeItems;
+    }
+
+    get onDidAccept(): Event<void> {
+        return this.onDidAcceptEmitter.event;
+    }
+
+    get placeholder(): string | undefined {
+        return this._placeholder;
+    }
+    set placeholder(placeholder: string | undefined) {
+        this._placeholder = placeholder;
+    }
+
+    get onDidChangeActive(): Event<T[]> {
+        return this.onDidChangeActiveEmitter.event;
+    }
+
+    get onDidChangeSelection(): Event<T[]> {
+        return this.onDidChangeSelectionEmitter.event;
+    }
+
+    get onDidChangeValue(): Event<string> {
+        return this.onDidChangeValueEmitter.event;
+    }
+
+    get onDidTriggerButton(): Event<QuickInputButton> {
+        return this.onDidTriggerButtonEmitter.event;
+    }
+
+    get onDidHide(): Event<void> {
+        return this.onDidHideEmitter.event;
+    }
+
+    dispose(): void {
+        this.disposableCollection.dispose();
+    }
+
+    hide(): void {
+        this.dispose();
+    }
+
+    show(): void {
+        const selectItem = (item: T) => {
+            this.activeItems = [item];
+            this.onDidAcceptEmitter.fire(undefined);
+            this.onDidChangeSelectionEmitter.fire([item]);
+            this.onDidHideEmitter.fire(undefined);
+            this.dispose();
+        };
+        this.quickOpen.showQuickPick(this.items.map(item => item as T), {
+            // tslint:disable-next-line:no-any
+            onDidSelectItem(item: T | string): any {
+                if (typeof item !== 'string') {
+                    selectItem(item);
```

Comment:
what about a case when item is not defined? Should not fire did hide event at least?

Reply:
fixed